File: src/ARC/ARCPath.fs
module ARC.Path 

open ISA
open FileSystem

let [<Literal>] gitKeepFileName = ".gitkeep" 
let [<Literal>] assaysFolderName = "assays"
let [<Literal>] assayProtocolsFolderName = "protocols"
let [<Literal>] assayDatasetFolderName = "dataset"
let [<Literal>] AssayFileName = "isa.assay.xlsx"
let [<Literal>] StudyFileName = "isa.study.xlsx"
let [<Literal>] InvestigationFileName = "isa.investigation.xlsx"
let [<Literal>] assayReadmeFileName = "Readme.md"

//let assaySubFolderNames = [|assayDatasetFolderName;assayProtocolsFolderName|]

//let combineAssayFolderPath (assay : ISA.ArcAssay) = 
//    //Assay.getIdentifier assay
//    //|> Path.combine assaysFolderName
//    raise (System.NotImplementedException())

//let combineAssayProtocolsFolderPath (assay : ISA.ArcAssay) = 
//    let assayFolder = combineAssayFolderPath assay
//    Path.combine assayFolder assayProtocolsFolderName

//let combineAssayDatasetFolderPath (assay : ISA.ArcAssay) = 
//    let assayFolder = combineAssayFolderPath assay
//    Path.combine assayFolder assayDatasetFolderName

//let combineAssaySubfolderPaths (assay : ISA.ArcAssay) = 
//    let assayFolder = combineAssayFolderPath assay
//    assaySubFolderNames
//    |> Array.map (fun subfolderName -> Path.combine assayFolder subfolderName)

File: ts/ARCPath.ts
import { combine } from "./FileSystem/Path.js";
import { Assay } from "./ISA/Assay.js";
import { map } from "./fable_modules/fable-library-ts/Array.js";

export const assaySubFolderNames: string[] = ["dataset", "protocols"];

export function combineAssayFolderPath(assay: Assay): string {
    return combine("assays", Assay.getIdentifier<string>(assay));
}

export function combineAssayProtocolsFolderPath(assay: Assay): string {
    return combine(combineAssayFolderPath(assay), "protocols");
}

export function combineAssayDatasetFolderPath(assay: Assay): string {
    return combine(combineAssayFolderPath(assay), "dataset");
}

export function combineAssaySubfolderPaths(assay: Assay): string[] {
    const assayFolder: string = combineAssayFolderPath(assay);
    return map<string, string>((subfolderName: string): string => combine(assayFolder, subfolderName), assaySubFolderNames);
}

